hub/core/lock.py

```diff
@@ -108,41 +146,59 @@ def release(self):
             terminate_thread(self._thread)
             self._acquired = False
         try:
-            del self.storage[get_dataset_lock_key()]
+            del self.storage[self.path]
         except Exception:
             pass
 
 
 _LOCKS: Dict[str, Lock] = {}
 
 
-def lock(storage: StorageProvider, callback: Optional[Callable] = None):
+def _get_lock_key(storage_path: str, version: Optional[str] = None):
```

Comment:
Call the argument commit_id instead. It should never be None, we can ignore the case